lib/hardware.go

```diff
@@ -0,0 +1,47 @@
+package lib
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/cybozu-go/well"
+)
+
+func DetectCPUNumber() (int, error) {
+	data, err := os.ReadFile("/proc/cpuinfo")
```

Comment:
You can iterate over lines by using [`bufio.Scanner`](https://pkg.go.dev/bufio#Scanner), without loading the whole data.
Though use of Scanner is not mandatory, it would be better to use a well-known idiom.

Reply:
It may be OK as it is because the next function uses the same iteration for a command output.
`bufio.Scanner` can be used also for a command output, but it would be overkill for a short output.